Refactor Code for Mass Order Unhold

AnshuMishra17 opened the pull request:
### Description
I have observed that MassAction Unhold is using the collection to release order from hold, whereas MassAction Hold is using Interface to put the order on hold.
So, I have refactor code in app/code/Magento/Sales/Controller/Adminhtml/Order/MassUnhold.php same as in app/code/Magento/Sales/Controller/Adminhtml/Order/MassHold.php

### Manual testing scenarios
<!--- Provide a set of unambiguous steps to test the proposed code change -->
1. Go to Sales -> Order
2. Select any number of orders (orders which are on hold.) 
3. Select Unhold Action from the Action dropdown.

AnshuMishra17 committed:
Refactor Code for Mass Order Unhold

I have observed that MassAction Unhold is using the collection to release order from hold, whereas MassAction Hold is using Interface to put the order on hold.
So, I have refactor code in app/code/Magento/Sales/Controller/Adminhtml/Order/MassUnhold.php same as in app/code/Magento/Sales/Controller/Adminhtml/Order/MassHold.php

magento-cicd2 commented:
[![CLA assistant check](https://cla.dev.magento.com/pull/badge/not_signed)](https://cla.dev.magento.com/magento/magento2?pullRequest=14629) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dev.magento.com/magento/magento2?pullRequest=14629) before we can accept your contribution.<br/>

sidolov commented on app/code/Magento/Sales/Controller/Adminhtml/Order/MassUnhold.php:
```diff
@@ -9,23 +9,31 @@
+    /**
+     * @var OrderManagementInterface
+     */
+    protected $orderManagement;
```
Please, make this property private

sidolov commented on app/code/Magento/Sales/Controller/Adminhtml/Order/MassUnhold.php:
```diff
@@ -9,23 +9,31 @@
+     * @param OrderManagementInterface $orderManagement
      */
-    public function __construct(Context $context, Filter $filter, CollectionFactory $collectionFactory)
+    public function __construct(Context $context, Filter $filter, CollectionFactory $collectionFactory, OrderManagementInterface $orderManagement)
```
Please, add new dependencies according to our [Backward Compatible Development Guide](http://devdocs.magento.com/guides/v2.2/contributor-guide/backward-compatible-development/#adding-a-constructor-parameter)

AnshuMishra17 replied:
I have added backward compatibility.

sidolov reviewed (changes requested):
@AnshuMishra17 , please, fix all issue according to review and take a look to failed tests

AnshuMishra17 committed:
Added backward compatibility.

Added backward compatibility for the new orderManagementInterface dependency and made the variable $orderManagement private.

Commit:
magento/magento2#14629 Fixed test failures

sidolov commented:
Hi @AnshuMishra17 , I fixed test failures

magento-engcom-team commented:
Hi @AnshuMishra17. Thank you for your contribution. 
Changes from your Pull Request will be available with the upcoming 2.2.5 release.
